extensions/cornerstone/src/registerCustomCornerstoneTools.ts

```diff
@@ -0,0 +1,132 @@
+import { toolNames } from './initCornerstoneTools';
+import { registerCustomToolsMappingFactory } from './utils/measurementServiceMappings/customToolsMappingFactory';
+import * as csTools from '@cornerstonejs/tools';
+
+const { BaseTool } = csTools;
+
+/**
+ * Object to store registered Custom Tools (classes)
+ */
+const customTools = {};
+
+/**
+ * Auxiliary class to store original customTools/toolNames to be restore on a fallback approach.
+ */
```

Comment:
Please add a lot more explanation here. I don't understand what is the reason for this Class? please explain one or two usecases